ui/hooks/useScrollRequired.js

```diff
@@ -11,7 +11,8 @@ import { debounce } from 'lodash';
  * @returns Flags for isScrollable and isScrollToBottom, a ref to use for the scrolling content, a scrollToBottom function and a onScroll handler.
  */
 export const useScrollRequired = (dependencies = []) => {
-  const ref = useRef();
+  const ref = useRef(null);
+  const [hasScrolledToBottomState, setHasScrolledToBottom] = useState(false);
```

Comment:
Why do we need both `hasScrolledToBottomState` and `isScrolledToBottomState`?

Reply:
unless I'm missing something... isScrolledToBottomState wasn't being reset when we scrolled up, so when I used this value to display our scroll-to-bottom button, it wasn't re-showing after the first time @FrederikBolding

Reply:
unless I'm missing something, isScrolledToBottomState wasn't being reset when we scrolled up, so when I used this value to display our scroll-to-bottom button, it wasn't re-showing after the first time @FrederikBolding

Reply:
now I'm seeing with this change, we'll want to apply it to update the other instances using this state 
https://metamask.github.io/metamask-storybook/?path=/docs/pages-snaps-snapconnect--docs

- essentially we'd use the `hasScrolledToBottom` state to enable the "Accept" button 
- as we can see in this current example, after scrolling up, the bottom no longer appears. We can use `isScrolledToBottom` for this

I can update these in this PR if you think this change looks good too

Reply:
on second thought... For these other instances, it looks okay since the content appears to be fairly short. Let me know if you'd like me to apply the change to show the button again after it disappears 👍🏼  cc: @FrederikBolding @eriknson 

current behavior:
https://github.com/MetaMask/metamask-extension/assets/20778143/53fccb3e-7a5b-4a43-a90d-2c0534c167b4